runtime/src/main/java/org/corfudb/runtime/view/LayoutBuilder.java

```diff
@@ -99,17 +99,17 @@ public LayoutBuilder removeUnresponsiveServer(@Nonnull String endpoint) {
      */
     public LayoutBuilder removeLayoutServer(String endpoint) {
 
-        List<String> layoutServers = layout.getLayoutServers();
```

Comment:
If we're going to change this function, I think we should properly document behavior.

First, endpoint should have a @Nonnull annotation.

Second, What's the behavior supposed to be if the endpoint is not in the layout? The current behavior is that the function silently succeeds. If that's what is desired, could you add that to the comment?

Third, add to the javadoc @throws that LayoutModificationException is thrown if it is the last remaining server.
 
That comment would make the code below more clear.

Reply:
@no2chem The change was intended to keep the behaviors as it was and just improve the internal process. However as you mentioned I think it is a good idea to change the API so that we do not allow null argument. I have done it for this and other public methods on the builder.

RE first point: **Done** (also done for other public methods)
RE seocond point: No changes in behavior happened here. your summarization is correct and adding comment is **Done**.
RE third point: **Done** also added comment for all the missing java docs.